### ISSUE TITLE
/sync takes *forever* to return if you fire up an old matrix client

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#9198]
it just took about 10 minutes for an incremental sync to return on my main account when firing up a client i hadn't used in a few months. surely incremental sync should be faster given it's dropping most of the data?

**Comment by bwindels** [comment]
any indication whether or not this was client or server-side? E.g. could you see how long the request took in the network tab?

**Comment by turt2live** [comment]
It's server side, and has always been slow in this scenario.

**Comment by bwindels** [comment]
Should we track this on the synapse side then? I suppose this is not https://github.com/matrix-org/synapse/issues/3880 as that's for initial sync?

**Comment by Half-Shot** [comment]
I'm actually finding this to be a problem for just 24 hours of inactivity. Obviously this is a condition of having a large account.

Probably what's missing from this issue is that the client's UX while it's trying to do the first incremental sync is quite poor. The room list doesn't render, and the client is full of blank spaces waiting for data. The displayname in the top left is an mxid. Overall, it's not clear to me why we don't just show a spinner while waiting for the first sync, as we do with initial syncs.

**Comment by novocaine** [comment]
> I'm actually finding this to be a problem for just 24 hours of inactivity. Obviously this is a condition of having a large account.
> 
> Probably what's missing from this issue is that the client's UX while it's trying to do the first incremental sync is quite poor. The room list doesn't render, and the client is full of blank spaces waiting for data. The displayname in the top left is an mxid. Overall, it's not clear to me why we don't just show a spinner while waiting for the first sync, as we do with initial syncs.

Agree. EW can't fix slow sync, but we should work out if the UI can be make as available as possible while the sync operation is in progress.
